omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -733,40 +515,56 @@ def _grasp(self, obj):
         grasp_poses = get_grasp_poses_for_object_sticky(obj)
         grasp_pose, object_direction = random.choice(grasp_poses)
 
+        grasp_offset_in_z = self.robot.finger_lengths[self.arm] + m.GRASP_APPROACH_DISTANCE
+
+        # Adjust grasp pose with reset orientation and finger length offset
+        reset_orientation = self._get_reset_eef_pose("world")[self.arm][1]
+        grasp_pos = grasp_pose[0] - object_direction * grasp_offset_in_z
+        grasp_quat = T.quat_multiply(grasp_pose[1], reset_orientation)
+        grasp_pose = (grasp_pos, grasp_quat)
+
         # Prepare data for the approach later.
-        approach_pos = grasp_pose[0] + object_direction * m.GRASP_APPROACH_DISTANCE
+        approach_pos = grasp_pose[0] + object_direction * grasp_offset_in_z
         approach_pose = (approach_pos, grasp_pose[1])
 
         # If the grasp pose is too far, navigate.
         indented_print("Navigating to grasp pose if needed")
         yield from self._navigate_if_needed(obj, pose_on_obj=grasp_pose)
 
         indented_print("Moving hand to grasp pose")
-        yield from self._move_hand(grasp_pose)
-
-        # We can pre-grasp in sticky grasping mode.
-        indented_print("Pregrasp squeeze")
-        yield from self._execute_grasp()
-
-        # Since the grasp pose is slightly off the object, we want to move towards the object, around 5cm.
-        # It's okay if we can't go all the way because we run into the object.
-        indented_print("Performing grasp approach")
-        yield from self._move_hand_linearly_cartesian(approach_pose, stop_on_contact=True)
-
-        # Step once to update
-        empty_action = self._empty_action()
-        yield self._postprocess_action(empty_action)
+        yield from self._move_hand(grasp_pose, motion_constraint=[0, 0, 0, 0, 1, 0])
```

Comment:
add comment to explain what does `motion_constraint=[0, 0, 0, 0, 1, 0]` do here.